### PR TITLE
musa: fix aarch64 build

### PR DESCRIPTION
**Pull request by BodhiHu**
Hi, this PR will:
1. fix building for MUSA when targeting `aarch64`;

### CODE REVIEW

**Comment by BodhiHu** [comment]
HI @ggerganov , could you please help review this PR ?

**Comment by ggerganov** [comment]
What errors do you get before this PR?

**Comment by BodhiHu** [comment]
Hi,

I got these errors(with the full compile command printed):

image

image

To fix this, had to undefine `__ARM_NEON` on the cuda code when `GGML_USE_MUSA` is on:
```
#ifdef GGML_USE_MUSA
#undef __ARM_NEON
#endif
```

**Comment by slaren** [comment]
The `-march` etc flags should not be used to build the musa backend. I would try fixing that first.

File: ggml/src/ggml-cpu/CMakeLists.txt [hunk 1]
```diff
@@ -169,6 +169,30 @@ function(ggml_add_cpu_backend_variant_impl tag_name)
                 # Raspberry Pi 3, 4, Zero 2 (32-bit)
                 list(APPEND ARCH_FLAGS -mno-unaligned-access)
             endif()
+            if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
+                message(STATUS "Linux aarch64 detected")
+
+                set(MARCH_FLAGS "-march=armv8.2-a+fp16")
```
Comment (kkontny) [inline ggml/src/ggml-cpu/CMakeLists.txt:175]: This assumes 2 things:
1. Every `aarch64` CPU supports fp16. I think the code should look for `fphp` CPU feature.
2. `aarch64` will be at least `armv8.2`, in fact `aarch64` implies only `armv8`

**Comment by BodhiHu** [comment]
Hi @slaren , @kkontny , I had removed the CPU parts and `-march` on musa side, to just fix the compiler errors.

**Comment by BodhiHu** [comment]
closed - as I think musa sdk should support arm neon.

**Pull request closed without merging**